src/app.tsx

```diff
@@ -142,6 +180,44 @@ export default function Chat() {
     agentMessages.length > 0 && scrollToBottom();
   }, [agentMessages, scrollToBottom]);
 
+  // Check for admin secret verification from tool results
+  useEffect(() => {
+    if (!adminSecret && agentMessages.length > 0) {
+      // Look for successful setAdminSecret tool invocation
+      for (const message of agentMessages) {
+        if (message.parts) {
+          for (const part of message.parts) {
+            if (
+              part.type === "tool-invocation" &&
+              part.toolInvocation.toolName === "setAdminSecret" &&
+              part.toolInvocation.state === "result"
+            ) {
+              const result = part.toolInvocation.result;
+              try {
+                // Parse JSON response
+                const parsedResult = JSON.parse(result);
+                if (parsedResult.status === "SUCCESS" && parsedResult.secret) {
+                  setAdminSecret(parsedResult.secret);
+                  sessionStorage.setItem(
+                    "pdf-admin-secret",
+                    parsedResult.secret
+                  );
+                  return;
+                }
+              } catch (error) {
+                // If JSON parsing fails, ignore this result
```

Comment:
this is not a useful comment plus it's kind of wrong since it's around more than the json parse - i'd just remove